app/views/posts/index.html.erb

```diff
@@ -1,44 +1,72 @@
 <h1>Timeline</h1>
 
+
 <ul class="index-list">
+
   <% @posts.each do |post| %>
-    <li class="home-li">
-      <div class="gravatar-list">   
-        <%= gravatar_for(post.user) %>
-        <div class= "name-posted-list"> 
-        <p><%= link_to post.user.name, post.user %> <%= link_to post.user.lastname, post.user %></p>
-        <p class="time"><%= time_ago_in_words(post.created_at) %> ago. </p>
-        </div> 
-      </div>
-   
-    <p class= "text-left timeline-content"><%= post.content%></p>
-    <p class="like-content">
-    <svg version="1.0" xmlns="http://www.w3.org/2000/svg"
-          width="1.1em" height="1em" viewBox="0 0 1200.000000 1419.000000"
-          preserveAspectRatio="xMidYMid meet">
+    <ul class="d-flex flex-column align-items-center justify-content-center">
+      <% if isit_me?(post.user) || friends_with?(post.author)%>
```

Comment:
Even if it's solved the problem , it's not the good way to do it
You are giving a work on the application instead of giving it to the database.
In the User model, you can add
```ruby
def friends_and_own_posts
  Post.where(user: (self.friends + self))
  # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
end
```
With that you are sure that the posts handled in the view are the only needed